Use full released xsum dataset

jbragg opened the pull request:
#672 Fix xsum to expand coverage and include IDs
Code based on parser from older version of `datasets/xsum/xsum.py`
@lhoestq

lhoestq commented on datasets/xsum/xsum.py:
```diff
@@ -73,39 +96,59 @@ def _info(self):
         """Returns SplitGenerators."""
 
-        dl_path = dl_manager.download_and_extract(_URL)
+        data_path = dl_manager.download_and_extract(_URL_DATA)
+        splits_path = dl_manager.download(_URL_SPLITS)
```
Could you change this to
```python
files_to_download = {
    "data": _URL_DATA,
    "splits": _URL_SPLITS
}
downloaded_files = dl_manager.download_and_extract(files_to_download)
```
since the download manager is able to do parallel downloads when a dict is passed.
Moreover it should fix the dummy data error in the CI

lhoestq commented on datasets/xsum/xsum.py:
```diff
@@ -73,39 +96,59 @@ def _info(self):
+        data_path = dl_manager.download_and_extract(_URL_DATA)
+        splits_path = dl_manager.download(_URL_SPLITS)
+
+        with open(splits_path, "r") as f:
```
Could you specify the encoding each time you call `open` ? `encoding="utf-8"` for example.
This is for windows users.

lhoestq reviewed (commented):
Thank you for updating xsum to get the IDS ! Also it looks like it also includes samples that were missing.

I left a few comments that should help you make the CI pass.

Moreover could you increase the dataset `VERSION` field ?
And remove the `SUPPORTED_VERSIONS` one that is not supported anyways.

jbragg commented:
@lhoestq I took a shot at addressing your comments but the build scripts seem to be complaining about not being able to open dummy files. How do I resolve those errors without copying the full dataset into the dummy dir?

lhoestq commented:
Could you check that the names of the dummy data files are right ?
You can use 
```
datasets-cli dummy_data ./datasets/xum
```
to print the expected file names

jbragg commented:
I did what it suggested
```
- Please create the following dummy data files 'dummy_data/XSum-TRAINING-DEV-TEST-SPLIT-90-5-5.json' from the folder './datasets/xsum/dummy/1.2.0'

- For each of the splits 'train, validation, test', make sure that one or more of the dummy data files provide at least one example

- If the method `_generate_examples(...)` includes multiple `open()` statements, you might have to create other files in addition to 'dummy_data/XSum-TRAINING-DEV-TEST-SPLIT-90-5-5.json'. In this case please refer to the `_generate_examples(...)` method
```
but this last point wasn't clear to me

jbragg commented:
Ok @lhoestq looks like I got the tests to pass :)

lhoestq reviewed (approved):
Awesome thank you !
